functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,293 @@ public static void composeObject_test6() throws Exception {
     }
   }
 
+  /** Test: enableObjectLegalHold(String bucketName, String objectName, String versionId) */
+  public static void enableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)");
+    }
+    long startTime = System.currentTimeMillis();
+    String bucketNameLegalHold = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(bucketNameLegalHold, null, true);
+      try (final InputStream is = new ContentInputStream(1 * KB)) {
+        client.putObject(bucketNameLegalHold, objectName, is, new PutObjectOptions(1 * KB, -1));
+      }
+
+      // Enable object legal hold.
+      client.enableObjectLegalHold(bucketNameLegalHold, objectName, "");
+
+      if (!client.isObjectLegalHoldEnabled(bucketNameLegalHold, objectName, "")) {
+        System.out.println(
+            "FAILED. expected Object Legal Hold Enabled to be : "
+                + "True"
+                + ", got: "
+                + client.isObjectLegalHoldEnabled(bucketNameLegalHold, objectName, ""));
+      }
+
+      try {
+        try (final InputStream is = new ContentInputStream(1 * KB)) {
+          client.putObject(bucketNameLegalHold, objectName, is, new PutObjectOptions(1 * KB, -1));
+        }
+      } catch (ErrorResponseException e) {
+        if (!e.errorResponse()
+            .message()
+            .equals("Object is WORM protected and cannot be overwritten")) {
+          mintFailedLog(
+              "enableObjectLegalHold",
+              "(String bucketName, String objectName, String versionId)",
+              startTime,
+              null,
+              e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+          throw e;
+        }
+      }
+      try {
+        client.removeObject(bucketNameLegalHold, objectName);
+      } catch (ErrorResponseException e) {
+        if (!e.errorResponse()
+            .message()
+            .equals("Object is WORM protected and cannot be overwritten")) {
+          mintFailedLog(
+              "enableObjectLegalHold",
+              "(String bucketName, String objectName, String versionId)",
+              startTime,
+              null,
+              e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
+          throw e;
+        }
+      }
+
+      client.disableObjectLegalHold(bucketNameLegalHold, objectName, "");
+      client.removeObject(bucketNameLegalHold, objectName);
+      client.removeBucket(bucketNameLegalHold);
```

Comment:
this cleanup won't be executed if one of above conditions throw exceptions. you would need to fix it accordingly.

Reply:
added finally block for cleanup